src/Optimizer/Transformer/PreloadHeroImage.php

```diff
@@ -717,4 +733,32 @@ private function supportsSrcset()
     {
         return $this->configuration->get(PreloadHeroImageConfiguration::PRELOAD_SRCSET);
     }
+
+    /**
+     * Check if an element or its ancestors is marked as a hero image.
+     *
+     * @param Element $element Element to check.
+     * @return bool Whether the element or one of its ancestors is marked as a hero image.
+     */
+    private function isMarkedAsHeroImage(Element $element)
+    {
+        while($element) {
+            if (!$element instanceof Element) {
+                $element = $element->parentNode;
+                continue;
+            }
+
+            if ($element->hasAttribute(Attribute::DATA_HERO)) {
+                return true;
+            }
```

Comment:
Since `DetermineHeroImages` in the AMP plugin is making educated decisions about what images are heroes in WordPress, I think such images should _not_ get `loading=lazy`.

```suggestion
            if ($element->hasAttribute(Attribute::DATA_HERO) || $element->hasAttribute(Attribute::DATA_HERO_CANDIDATE)) {
                return true;
            }
```

Reply:
Or maybe not per https://github.com/ampproject/amp-wp/pull/6062/files#r611001630.  Alternatively: https://github.com/ampproject/amp-toolbox-php/pull/141/files#r611001917.